transport/grpc/headers.go

```diff
@@ -103,6 +107,7 @@ func transportRequestToMetadata(request *transport.Request) (metadata.MD, error)
 		addToMetadata(md, RoutingKeyHeader, request.RoutingKey),
 		addToMetadata(md, RoutingDelegateHeader, request.RoutingDelegate),
 		addToMetadata(md, EncodingHeader, string(request.Encoding)),
+		addToMetadata(md, CallerProcedureHeader, request.CallerProcedure),
```

Comment:
I feel we should add the callerProcedureHeader only if the request has non-empty callerProcedure, wdyt?

Reply:
Done